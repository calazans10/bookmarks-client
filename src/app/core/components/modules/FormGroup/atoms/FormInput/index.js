import React from 'react';
import PropTypes from 'prop-types';
import InputMask from 'react-input-mask';

const FormInput = ({
  input,
  type,
  mask,
  formatChars,
  autoFocus,
  disabled,
  placeholder,
  onFocus,
  onBlur,
}) => (
  <InputMask
    {...input}
    id={input.name}
    className="form-group__control"
    type={type}
    mask={mask}
    formatChars={formatChars}
    maskChar={null}
    onFocus={onFocus}
    onBlur={onBlur}
    autoFocus={autoFocus}
    disabled={disabled}
    placeholder={placeholder}
  />
);

FormInput.propTypes = {
  input: PropTypes.object.isRequired,
  type: PropTypes.string.isRequired,
  mask: PropTypes.string.isRequired,
  formatChars: PropTypes.object.isRequired,
  autoFocus: PropTypes.bool.isRequired,
  disabled: PropTypes.bool.isRequired,
  placeholder: PropTypes.string.isRequired,
  onBlur: PropTypes.func.isRequired,
  onFocus: PropTypes.func.isRequired,
};

export default FormInput;
