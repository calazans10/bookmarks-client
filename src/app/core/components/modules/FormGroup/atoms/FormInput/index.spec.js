import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import FormInput from './index';

describe('FormInput', () => {
  it('renders without crashing', () => {
    const props = {
      input: {
        name: 'test',
        value: '',
      },
      type: 'text',
      mask: '',
      formatChars: { 9: '[0-9]', '?': '[0-9]' },
      onFocus: jest.fn(),
      onBlur: jest.fn(),
      autoFocus: false,
      disabled: false,
      placeholder: 'A placeholder',
    };
    const wrapper = shallow(<FormInput {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
