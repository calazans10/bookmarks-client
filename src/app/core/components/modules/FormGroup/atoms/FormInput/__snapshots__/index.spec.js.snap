// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput renders without crashing 1`] = `
<InputElement
  autoFocus={false}
  className="form-group__control"
  disabled={false}
  formatChars={
    Object {
      "9": "[0-9]",
      "?": "[0-9]",
    }
  }
  id="test"
  mask=""
  maskChar={null}
  name="test"
  onBlur={[MockFunction]}
  onFocus={[MockFunction]}
  placeholder="A placeholder"
  type="text"
  value=""
/>
`;
